resources/table/pagination.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { FaChevronLeft, FaChevronRight } from "react-icons/fa6";
+
+interface PaginationProps {
+  currentPage: number;
+  totalPages: number;
+  totalRecords: number;
+  onPageChange: (page: number) => void;
+}
+
+const Pagination: React.FC<PaginationProps> = ({ currentPage, totalPages, totalRecords, onPageChange }) => {
+  const maxPageNumbersToShow = 5; // Maximum number of page numbers to show at once
+  const halfMax = Math.floor(maxPageNumbersToShow / 2);
+  let startPage = Math.max(1, currentPage - halfMax);
+  let endPage = Math.min(totalPages, currentPage + halfMax);
+
+  if (endPage - startPage + 1 < maxPageNumbersToShow) {
+    if (startPage === 1) {
+      endPage = Math.min(totalPages, startPage + maxPageNumbersToShow - 1);
+    } else if (endPage === totalPages) {
+      startPage = Math.max(1, endPage - maxPageNumbersToShow + 1);
+    }
+  }
+
+  const pageNumbers = [];
+  for (let i = startPage; i <= endPage; i++) {
+    pageNumbers.push(i);
+  }
+
+  return (
+    <div className="flex justify-between items-center mt-4 text-gray-500">
+      <div>
+        Total Records: {totalRecords}, Page {currentPage} of {totalPages}
+      </div>
+      <div className="flex items-center space-x-2">
+        <button
+          onClick={() => onPageChange(Math.max(currentPage - 1, 1))}
+          disabled={currentPage === 1}
+          className="px-4 py-2 bg-gray-200 rounded disabled:opacity-50"
```

Comment:
Lets grapp the button in a PaginatedButton in  a sub function in the same page.

You can left paddings outside the component and keep them in this level.

Reply:
> You can left paddings outside the component and keep them in this level.

I didn't fully understand this line what you mean here? :D